src/frontend/js/types/Suggestion.ts

```diff
@@ -0,0 +1,95 @@
+import { FormattedMessage } from 'react-intl';
+
+import { CategoryForSuggestion } from './Category';
+import { CourseForSuggestion } from './Course';
+import { modelName } from './models';
+import { OrganizationForSuggestion } from './Organization';
+
+/**
+ * A Default suggestion shape for cases where we need one.
+ */
+interface DefaultSuggestion {
+  kind: 'default';
+  title: string;
+}
+
+/**
+ * A suggestion shape for Category autocomplete.
+ */
+interface CategorySuggestion {
+  kind: modelName.CATEGORIES;
+  data: CategoryForSuggestion;
+}
+
+/**
+ * A suggestion shape for Course autocomplete.
+ */
+interface CourseSuggestion {
+  kind: modelName.COURSES;
+  data: CourseForSuggestion;
+}
+
+/**
+ * A suggestion shape for Organization autocomplete.
+ */
+interface OrganizationSuggestion {
+  kind: modelName.ORGANIZATIONS;
+  data: OrganizationForSuggestion;
+}
+
+/**
+ * Utility type that allows a consumer to easily define the kinds of suggestions it supports and
+ * get proper typechecking around them.
+ */
+export type Suggestion<Kind> = Kind extends modelName.CATEGORIES
```

Comment:
I agree this would look great with a switch, unfortunately, `switch` is not an expression in JS/TS and we cannot use it here. As far as I know the prettier line break on a ternary is the best we can get

Reply:
Unfortunately this is what I thought